[17.0][MIG] mrp_lot_number_propagation

astirpe opened the pull request.

sebalix committed:
[ADD] mrp_lot_number_propagation

Commit:
[UPD] Update mrp_lot_number_propagation.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] icon.png

mymage committed:
Added translation using Weblate (Italian)

grindtildeath committed:
Allow to propagate lot from multiple BOM lines for variants

When defining a single BOM for a product template having multiple
product variants, we can have different components where the lot
number must be propagated for different product variants.

Therefore we need to allow to mark multiple BOM line with
propagate_lot_number, and to avoid complicating the check function
on the BOM lines, we ensure at the manufacturing order confirmation
that only a single component is set to propagate its lot number.

grindtildeath committed:
mrp_lot_number_propagation: Add hook to get propagating component

OCA-git-bot committed:
mrp_lot_number_propagation 15.0.0.1.1

Commit:
[UPD] Update mrp_lot_number_propagation.pot

OCA-git-bot committed:
mrp_lot_number_propagation 15.0.0.2.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: manufacture-15.0/manufacture-15.0-mrp_lot_number_propagation
Translate-URL: https://translation.odoo-community.org/projects/manufacture-15-0/manufacture-15-0-mrp_lot_number_propagation/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (23 of 23 strings)

Translation: manufacture-15.0/manufacture-15.0-mrp_lot_number_propagation
Translate-URL: https://translation.odoo-community.org/projects/manufacture-15-0/manufacture-15-0-mrp_lot_number_propagation/it/

sebalix committed:
mrp_lot_number_propagation: re-use existing lot

Commit:
[UPD] Update mrp_lot_number_propagation.pot

OCA-git-bot committed:
mrp_lot_number_propagation 15.0.0.3.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: manufacture-15.0/manufacture-15.0-mrp_lot_number_propagation
Translate-URL: https://translation.odoo-community.org/projects/manufacture-15-0/manufacture-15-0-mrp_lot_number_propagation/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (24 of 24 strings)

Translation: manufacture-15.0/manufacture-15.0-mrp_lot_number_propagation
Translate-URL: https://translation.odoo-community.org/projects/manufacture-15-0/manufacture-15-0-mrp_lot_number_propagation/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (24 of 24 strings)

Translation: manufacture-15.0/manufacture-15.0-mrp_lot_number_propagation
Translate-URL: https://translation.odoo-community.org/projects/manufacture-15-0/manufacture-15-0-mrp_lot_number_propagation/it/

bosd committed:
[MIG] mrp_lot_number_propagation: Migration to 16.0

bosd committed:
mrp_lot_number_propagation: Promote to Beta development status

Commit:
[UPD] Update mrp_lot_number_propagation.pot

OCA-git-bot committed:
[BOT] post-merge updates

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (24 of 24 strings)

Translation: manufacture-16.0/manufacture-16.0-mrp_lot_number_propagation
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-mrp_lot_number_propagation/es/

CasVissers-360ERP commented on mrp_lot_number_propagation/models/mrp_production.py:
```diff
@@ -0,0 +1,174 @@
+        for order in self:
+            if (
+                order.is_lot_number_propagated
+                and "lot_producing_id" in vals
```
@astirpe if lot_producing_id is False we should not raise the error

astirpe replied:
@CasVissers-360ERP done here https://github.com/OCA/manufacture/pull/1299/commits/c62f997d23aaf9b98f80a10058f22c88a6317458 could you check?

CasVissers-360ERP replied:
Thnx resolved.

bosd commented:
@astirpe Can you please squash your last commit?

astirpe commented:
@bosd squashed, thanks!

bosd commented:
Seems like the @OCA-git-bot is still on holiday, as this one carries the approved label and can be merged.

Just noticed the "confusing" naming of this module.
Technical name is lot propagation, but the title reads serial propagation.
![image](https://github.com/user-attachments/assets/3e68d181-c593-416d-a8b1-292df6212502)

Highcooley commented:
> Seems like the @OCA-git-bot is still on holiday, as this one carries the approved label and can be merged.
> 
> Just noticed the "confusing" naming of this module. Technical name is lot propagation, but the title reads serial propagation. ![image](https://private-user-images.githubusercontent.com/11499387/370748943-3e68d181-c593-416d-a8b1-292df6212502.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk3NjIyNjIsIm5iZiI6MTcyOTc2MTk2MiwicGF0aCI6Ii8xMTQ5OTM4Ny8zNzA3NDg5NDMtM2U2OGQxODEtYzU5My00MTZkLWE4YjEtMjkyZGY2MjEyNTAyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI0VDA5MjYwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAyZjJiOTQ0MDk5NmNhZTY0ZTNiOTk5OWViMzI5Njc2Y2Y5NDE3OTExMjliMThjMDkxOGEzMDU1ZDYyNTdmNjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.rni7dtD5V7mpc7AQAbqFtDdKg85X5rX-QM4BJb42diE)

The idea was to leave this option open in the technical name to also propagate LOT numbers in the future and not just SN. Also Odoo std naming only talks about LOT and not about SN.

bosd commented:
Ok for me, Let's merge!

bosd committed:
[FIX] mrp_lot_number_propagation Keyerror stock.production.lot

bosd committed:
[FIX] mrp_lot_number_propagation: tests

import test_mrp_production
stock.move.line : product_uom_qty -> reserved_uom_qty

bosd reviewed (changes requested):
Tests are not completely run.
Please cherry pick 0762c0a9ea9cbedc25b072d8a034caae83573348 from #1437

astirpe committed:
[MIG] mrp_lot_number_propagation: pre-commit

astirpe committed:
[MIG] mrp_lot_number_propagation: migrate to V17

astirpe commented:
@bosd done, thank you!

bosd commented:
@astirpe Thanks,
I just noticed the "Display Display Propagate Lot Number" field in the view of the BoM. 
Don't recall this from previous versions. Does the field do something? Is this intended?

Without Propagate box thicked:
![image](https://github.com/user-attachments/assets/d9630492-2f71-488a-bab6-fddc8ce8a5c7)

with:
![image](https://github.com/user-attachments/assets/44776f5f-1566-4fc2-bca6-82a4a26bb6be)

astirpe commented:
@bosd I think you are right, I don't see anything making use of "Display Propagate Lot Number" field. It was there since the first release of the module though.

sebalix commented:
@bosd @astirpe I don't remember neither, probably dead code from a first implementation. I could not find any specific code neither that could make use of this field on `mrp.bom.line`, so I would say you can remove it :+1:

bosd reviewed (approved):
Approving, as it should not block this pr. The issue is in multiple version. Better to have it in a separate commit for cherry-pciking.

StefanRijnhart reviewed (approved):
/ocabot merge nobump

StefanRijnhart commented:
@astirpe Please clean outdated rst readme fragments manually

grindtildeath reviewed (approved):
@astirpe Thanks for the mig. Can you please fix the build so we can merge? :pray:
